src/commands/react_native/xcode.rs

```diff
@@ -371,21 +365,44 @@ pub fn execute(matches: &ArgMatches) -> Result<()> {
                 chunk_upload_options: chunk_upload_options.as_ref(),
             })?;
         } else {
-            let dist = dist_from_env.unwrap_or_else(|_| plist.build().to_string());
-            let release_name = release_from_env.unwrap_or(format!(
-                "{}@{}+{}",
-                plist.bundle_id(),
-                plist.version(),
-                dist
-            ));
+            let (dist, release_name) = match (dist_from_env, release_from_env) {
+                (Ok(dist_env), Ok(release_env)) => {
+                    // Both environment variables are present
+                    (Some(dist_env), Some(release_env))
+                },
+                (Ok(dist_env), Err(_)) => {
+                    // Only dist environment variable is present
+                    (Some(dist_env), None)
+                },
+                (Err(_), Ok(release_env)) => {
+                    // Only release environment variable is present
+                    (None, Some(release_env))
+                },
+                (Err(_), Err(_)) => {
+                    // Neither environment variable is present, attempt to parse Info.plist
+                    info!("Parsing Info.plist");
+                    match InfoPlist::discover_from_env() {
+                        Ok(Some(plist)) => {
+                            // Successfully discovered and parsed Info.plist
+                            let dist_string = plist.build().to_string();
+                            let release_string = format!("{}@{}+{}", plist.bundle_id(), plist.version(), dist_string);
+                            info!("Parse result from Info.plist: {:?}", &plist);
+                            (Some(dist_string), Some(release_string))
+                        },
+                        _ => {
+                            bail!("Info.plist was not found or an parsing error occurred");
+                        }
+                    }
+                }
```

Comment:
```suggestion
                (Err(_), Err(_)) => {
                    // Neither environment variable is present, attempt to parse Info.plist
                    info!("Parsing Info.plist");
                    match InfoPlist::discover_from_env() {
                        Ok(Some(plist)) => {
                            // Successfully discovered and parsed Info.plist
                            let dist_string = plist.build().to_string();
                            let release_string = format!("{}@{}+{}", plist.bundle_id(), plist.version(), dist_string);
                            info!("Parse result from Info.plist: {:?}", &plist);
                            (Some(dist_string), Some(release_string))
                        },
                        _ => {
                            bail!("Info.plist was not found or an parsing error occurred");
                        }
                    }
                }
                _ => (dist_from_env.ok(), release_from_env.ok()),
```

We can combine the first three cases into a single case by using the [`Result::ok`](https://doc.rust-lang.org/std/result/enum.Result.html#method.ok) function, which converts `Result<T, E>` into `Option<T>`. To do this, however, the order needs to be rearranged.